api/src/services/observation-service.ts

```diff
@@ -556,7 +560,7 @@ export class ObservationService extends DBService {
 
     const measurementsToValidate: IMeasurementDataToValidate[] = worksheetRowObjects.flatMap((row) => {
       return measurementColumnNames.map((columnName) => ({
-        tsn: String(row['ITIS_TSN'] ?? row['TSN'] ?? row['TAXON'] ?? row['SPECIES']),
+        tsn: String(getColumnCellValue(row, 'ITIS_TSN').cell),
```

Comment:
String(undefined) will lead to unexpected results -> 'undefined'. Are we sure this value is defined at this point?

Reply:
Yes, if ```getColumnCellValue(row, 'ITIS_TSN').cell``` returns undefined, the ```validateCsvFile``` function on line 515 would throw an error.